mods/ra/maps/allies-06a/allies06a.lua

```diff
@@ -95,94 +95,95 @@ InitialSovietPatrols = function()
 end
 
 InitialAlliedReinforcements = function()
-	local camera = Actor.Create("Camera", true, { Owner = player, Location = DefaultCameraPosition.Location })
+	local camera = Actor.Create("Camera", true, { Owner = Greece, Location = DefaultCameraPosition.Location })
 	Trigger.AfterDelay(DateTime.Seconds(30), camera.Destroy)
 
-	Reinforcements.Reinforce(player, AlliedReinforcementsA, { AlliedEntry1.Location, UnitBStopLocation.Location }, 2)
+	Reinforcements.Reinforce(Greece, AlliedReinforcementsA, { AlliedEntry1.Location, UnitBStopLocation.Location }, 2)
 	Trigger.AfterDelay(DateTime.Seconds(2), function()
-		Reinforcements.Reinforce(player, AlliedReinforcementsB, { AlliedEntry2.Location, UnitAStopLocation.Location }, 2)
+		Reinforcements.Reinforce(Greece, AlliedReinforcementsB, { AlliedEntry2.Location, UnitAStopLocation.Location }, 2)
 	end)
 	Trigger.AfterDelay(DateTime.Seconds(5), function()
-		Reinforcements.Reinforce(player, { "mcv" }, { AlliedEntry3.Location, MCVStopLocation.Location })
+		Reinforcements.Reinforce(Greece, { "mcv" }, { AlliedEntry3.Location, MCVStopLocation.Location })
 	end)
 end
 
 CaptureRadarDome = function()
 	Trigger.OnKilled(RadarDome, function()
-		player.MarkFailedObjective(CaptureRadarDomeObj)
+		Greece.MarkFailedObjective(CaptureRadarDomeObj)
 	end)
 
 	Trigger.OnCapture(RadarDome, function()
-		player.MarkCompletedObjective(CaptureRadarDomeObj)
+		Greece.MarkCompletedObjective(CaptureRadarDomeObj)
 
+		--- @param a actor
```

Comment:
This is also kind of weird.

Reply:
Do you want it removed?

Reply:
If we can yes.

Reply:
What do you mean? If I remove this, the project will have warnings.

Reply:
Why isn't this part of the Utils.Do specification? We know what the type of the parameter in the callback is. (The type of the elements of the input table.) Why is this needed just for this single Utils.Do?

Reply:
~~It would be great to figure out how to specify that on `Utils.Do`~~
Wait, isn't `Utils.Do` supposed to be super generic? If we only use it on actor collections that's just us having a limited use-case.
The C# signature is `void Do(LuaValue[] collection, LuaFunction func)`
and the Lua signature is `Do = function(collection, func) end;`
The argument of the `func` would be whatever isin the `collection`, but I don't know how we could specify that in the Lua API (the one generated for the vscode extension).

And this is needed here and not in our other uses of `Do` because this, by adding a `CVec` to an `unknown` and trying to assign the result to a `CPos` gives us `Cannot assign 'number' to 'cpos'.` The other uses apparently don't care as much.

Reply:
> Wait, isn't Utils.Do supposed to be super generic?

Yes it is. Hence why this should be solved on a general level.

Reply:
Removed all workarounds.